### ISSUE TITLE
Add validations for gpx upload

### ISSUE THREAD

**Issue by tomhughes** [tomhughes/trac-tickets#1510]
**[Submitted to the original trac issue database at 9.35pm, Sunday, 18th January 2009]**

In create:
Check for user on upload.

In api_create:
Check that the file is provided.

**Comment by tomhughes** [comment]
**Author: edgemaster**
**[Added to the original trac issue at 1.19am, Friday, 23rd January 2009]**

(In [13321]) Require auth on calls to /trace/create, and pass through to form if called without params. Adjust validations on traces to prevent dual error messages on description (validates_presence_of catches the empty string removed from length validation)
More changes to come for api_create
References #1510

**Comment by tomhughes** [comment]
**Author: edgemaster**
**[Added to the original trac issue at 7.14pm, Friday, 23rd January 2009]**

(In [13331]) Set params to sane defaults if unset by client.
Reject creates with no file with 400, model validations will also reject with 400 (as was existing but broken behaviour)
Closes #1510

**Comment by tomhughes** [comment]
**Author: grand.edgemaster@gmail.com**
**[Added to the original trac issue at 7.28pm, Friday, 23rd January 2009]**

For the context of any future reviews errors such as the following would occur without the appropriate parameters to api_create:
```
NoMethodError (You have a nil object when you didn't expect it!
You might have expected an instance of Array.
The error occurred while evaluating nil.split):
    /app/models/trace.rb:27:in `tagstring='
```
